src/transformers/models/textnet/modeling_textnet.py

```diff
@@ -0,0 +1,552 @@
+        running_mean = batch_norm.running_mean
+        running_var = batch_norm.running_var
+        gamma = batch_norm.weight
+        beta = batch_norm.bias
+        eps = batch_norm.eps
+        std = (running_var + eps).sqrt()
+        t = (gamma / std).reshape(-1, 1, 1, 1)
+        return kernel * t, beta - running_mean * gamma / std
+
+    def get_equivalent_kernel_bias(self):
+        kernel_mxn, bias_mxn = self._fuse_batch_norm_tensor(self.main_conv, self.main_batch_norm)
+        if self.vertical_conv is not None:
+            kernel_mx1, bias_mx1 = self._fuse_batch_norm_tensor(self.vertical_conv, self.vertical_batch_norm)
+        else:
+            kernel_mx1, bias_mx1 = 0, 0
+        if self.horizontal_conv is not None:
+            kernel_1xn, bias_1xn = self._fuse_batch_norm_tensor(self.horizontal_conv, self.horizontal_batch_norm)
+        else:
+            kernel_1xn, bias_1xn = 0, 0
+        kernel_id, bias_id = self._identity_to_conv(self.rbr_identity)
+        kernel_mxn = kernel_mxn + kernel_mx1 + kernel_1xn + kernel_id
+        bias_mxn = bias_mxn + bias_mx1 + bias_1xn + bias_id
+        return kernel_mxn, bias_mxn
+
+    def _pad_to_mxn_tensor(self, kernel):
+        kernel_height, kernel_width = self.kernel_size
+        height, width = kernel.shape[2:]
+        pad_left_right = (kernel_width - width) // 2
+        pad_top_down = (kernel_height - height) // 2
+        return torch.nn.functional.pad(kernel, [pad_left_right, pad_left_right, pad_top_down, pad_top_down])
+
+    def prepare_for_eval(self):
+        kernel, bias = self.get_equivalent_kernel_bias()
+        self.fused_conv = nn.Conv2d(
+            in_channels=self.main_conv.in_channels,
+            out_channels=self.main_conv.out_channels,
+            kernel_size=self.main_conv.kernel_size,
+            stride=self.main_conv.stride,
+            padding=self.main_conv.padding,
+            bias=True,
+        )
+        self.fused_conv.weight.data = kernel
+        self.fused_conv.bias.data = bias
+        for para in self.fused_conv.parameters():
+            para.detach_()
+
+
+class TextNetPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = TextNetConfig
+    base_model_prefix = "textnet"
+    main_input_name = "pixel_values"
+
+    def _init_weights(self, module):
+        if isinstance(module, (nn.Linear, nn.Conv2d)):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+
+
+@add_start_docstrings(
+    "The bare Textnet model outputting raw features without any specific head on top.",
+    TEXTNET_START_DOCSTRING,
+)
+class TextNetModel(TextNetPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.first_conv = TextNetConvLayer(
```

Comment:
```suggestion
        self.stem = TextNetConvLayer(
```
we usually call this the stem